iroh-metrics/src/config.rs

```diff
@@ -0,0 +1,48 @@
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// The name of the service. Should be the same as the Cargo package name.
+    pub service_name: String,
+    /// A unique identifier for this instance of the service.
+    pub instance_id: String,
+    /// The build version of the service (commit hash).
+    pub build: String,
+    /// The version of the service. Should be the same as the Cargo package version.
+    pub version: String,
+    /// The environment of the service.
+    pub service_env: String,
+    /// Flag to enable debug mode.
+    pub debug: bool,
+    /// The endpoint of the trace collector.
+    pub collector_endpoint: String,
+    /// The endpoint of the prometheus push gateway.
+    pub prometheus_gateway_endpoint: String,
+}
+
+impl Config {
+    pub fn new(
+        service_name: String,
```

Comment:
why not grab the `service_name` automatically here, instead of having the caller do it manually

Reply:
cant, `env!("CARGO_PKG_NAME").to_string()` resolves at compile time and is the value of the current crate. Putting it here would always resolve to `iroh-metrics` for all services